toolkit/tools/pkg/image/pkgfetcher/pkgfetcher.go

```diff
@@ -0,0 +1,123 @@
+package pkgfetcher
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/packagerepo/repocloner"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/packagerepo/repoutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkggraph"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/pkgjson"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/pkg/imagegen/configuration"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/pkg/imagegen/installutils"
+)
+
+func FetchPkgsAndCreateRepo(cfg *Config) error {
+	if cfg.ExternalOnly && strings.TrimSpace(cfg.InputGraph) == "" {
+		logger.Log.Info("input-graph must be provided if external-only is set.")
+		return errors.New("input-graph must be provided")
+	}
+
+	cloner := rpmrepocloner.New()
+	err := cloner.Initialize(cfg.OutDir, cfg.TmpDir, cfg.WorkerTar, cfg.ExistingRpmDir, cfg.UsePreviewRepo, cfg.RepoFiles)
+	if err != nil {
+		logger.Log.Infof("Failed to initialize RPM repo cloner. Error: %s", err)
+		return err
+	}
+	defer cloner.Close()
+
+	if !cfg.DisableUpstreamRepos {
+		tlsKey, tlsCert := strings.TrimSpace(cfg.TlsClientKey), strings.TrimSpace(cfg.TlsClientCert)
+		err = cloner.AddNetworkFiles(tlsCert, tlsKey)
+		if err != nil {
+			logger.Log.Infof("Failed to customize RPM repo cloner. Error: %s", err)
+			return err
+		}
+	}
+
+	if strings.TrimSpace(cfg.InputSummaryFile) != "" {
+		// If an input summary file was provided, simply restore the cache using the file.
+		err = repoutils.RestoreClonedRepoContents(cloner, cfg.InputSummaryFile)
+	} else {
+		err = cloneSystemConfigs(cloner, cfg.ConfigFile, cfg.BaseDirPath, cfg.ExternalOnly, cfg.InputGraph)
+	}
+
+	if err != nil {
+		logger.Log.Infof("Failed to clone RPM repo. Error: %s", err)
```

Comment:
Why are errors logged on info level now? We can go down to error and make sure the caller does the right thing or keep the original changes, like what we did in other commits.